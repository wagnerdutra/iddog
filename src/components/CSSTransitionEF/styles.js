import styled from 'styled-components';

const Transition = styled.div`
  .effect-enter {
    opacity: 0;
  }
  .effect-enter-active {
    opacity: 1;
    transition: opacity 200ms;
  }
  .effect-exit {
    opacity: 1;
  }
  .effect-exit-active {
    opacity: 0;
    transition: opacity 200ms;
  }
`;

export { Transition };
