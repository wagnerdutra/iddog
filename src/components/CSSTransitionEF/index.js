import React from 'react';
import propTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';

import './styles.css';

const CSSTransitionEF = ({ children, inProp }) => (
  <CSSTransition in={inProp} timeout={300} classNames="effect" unmountOnExit>
    {children}
  </CSSTransition>
);

CSSTransitionEF.propTypes = {
  children: propTypes.node.isRequired,
  inProp: propTypes.bool.isRequired
};

export default CSSTransitionEF;
