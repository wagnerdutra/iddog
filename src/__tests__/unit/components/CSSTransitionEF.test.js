import React from 'react';

import { render } from 'react-testing-library';

import CSSTransitionEF from 'Components/CSSTransitionEF';

describe('<CSSTransitionEF />', () => {
  it('should render with show false correctly', () => {
    const { container } = render(
      <CSSTransitionEF inProp>
        <p>Teste</p>
      </CSSTransitionEF>
    );
    expect(container).toMatchSnapshot();
  });
});
