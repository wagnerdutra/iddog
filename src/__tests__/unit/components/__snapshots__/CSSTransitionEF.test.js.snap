// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CSSTransitionEF /> should render with show false correctly 1`] = `
<div>
  <p>
    Teste
  </p>
</div>
`;
